### PR TITLE
Features/widget test history InfoCard

### PR DESCRIPTION
**Pull request by krushnakantanayak**
### Release

1.0.0
https://blackminystudio.atlassian.net/browse/TPA-69

### Description

- Test Coverage:
- Valid inputs (date and amount).
- Empty and non-numeric amounts.
- Large numeric amounts (formatted correctly).
- Empty date.
- Theme styling compliance.
- Layout responsiveness for smaller screens.
- Visual elements like separator lines.

### Screenshots

history_info_card_test

### Tips

Any tips for testing it?

---

### IMPORTANT:

- (REQUIRED) Coverage Screenshots for the tests written
- (REQUIRED) UI Screenshots for the UI implementation written

### Author preparation checklist

- [x] Current branch is rebased with latest 'Staging' branch
- [x] PR is targeting only one issue
- [x] Commit messages are descriptive, consistent, and follow the project's guidelines.
- [ ] Branch name matches with at least `<jira link reference>-<short change description>`
- [x] Primary reviewer is set as assignee to the PR
- [x] No unnecessary changes on `pubspec.lock`
- [x] All added TODOs have associated Github issue references
- [x] \*.iml and, pubsepc_overrides.yaml are not checkedin
- [x] All hard coded strings are present in constants file

### Primary reviewer checklist

- [x] All items in the author checklist is checked or not applicable
- [x] All files including .lock and assets are reviewed
- [x] Reasonable UI flows/cases related to the change are manually tested
- [ ] No commented-out code or unnecessary debug statements.
- [x] No spelling or grammatical errors in code comments or documentation.
- [x] No sensitive data (e.g., API keys, credentials) is hardcoded or exposed.
- [x] The branch is up-to-date with the target branch (e.g., staging).
- [x] The code adheres to established coding standards.

### Pre-merge checklist

- [ ] Two approvals received. (Changes affecting automatic PR verification, like new linter rules, formatting rules, Github workflow configurations)

---

### CODE REVIEW

**Review by satyabratanayak — APPROVED** [review]

File: test/widgets/cards/history_info_card_test.dart [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+      final amountTextStyle = tester.widget<Text>(totalAmountTextFinder).style;
+
+      expect(dateTextStyle?.color, ColorTokens.contrastMedium);
+      expect(amountTextStyle?.color, ColorTokens.secondary);
+    });
+
+    testWidgets(
+        'Given an empty amount '
+        'when HistoryInfoCard is rendered '
+        'Then it should display only the rupee symbol without crashing',
```
Comment (quiteachaos) [inline test/widgets/cards/history_info_card_test.dart]: not just rupee symbol but ₹0 should be displayed
`Then it should display 0 with rupee symbol without crashing`

File: test/widgets/cards/history_info_card_test.dart [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+          date: testDate,
+          amount: '',
+        ),
+      );
+
+      // Assert
+      expect(find.text(rupeeSymbol), findsOneWidget);
+    });
+
+    testWidgets(
```
Comment (quiteachaos) [inline test/widgets/cards/history_info_card_test.dart]: combine this test with first test case

File: test/widgets/cards/history_info_card_test.dart [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+          ),
+        ),
+      );
+
+      // Assert
+      expect(find.text('$rupeeSymbol$formattedTestAmount'), findsOneWidget);
+      expect(find.text(testDate), findsOneWidget);
+    });
+
+    testWidgets(
```
Comment (quiteachaos) [inline test/widgets/cards/history_info_card_test.dart]: no need of this.. remove this test

File: lib/widgets/cards/history_info_card.dart [hunk 4]
```diff
@@ -88,6 +88,7 @@ class HistoryInfoCard extends StatelessWidget {
           mainAxisSize: MainAxisSize.min,
           crossAxisAlignment: CrossAxisAlignment.end,
           children: [
+            // TODO:renders correctly with empty amount
```
Comment (quiteachaos) [inline lib/widgets/cards/history_info_card.dart]: complete the TODO

File: test/widgets/cards/history_info_card_test.dart (hunk 1 above)
Comment (krushnakantanayak) [reply test/widgets/cards/history_info_card_test.dart]: done

File: test/widgets/cards/history_info_card_test.dart (hunk 2 above)
Comment (krushnakantanayak) [reply test/widgets/cards/history_info_card_test.dart]: done

File: lib/widgets/cards/history_info_card.dart (hunk 4 above)
Comment (krushnakantanayak) [reply lib/widgets/cards/history_info_card.dart]: done

File: test/widgets/cards/history_info_card_test.dart (hunk 3 above)
Comment (krushnakantanayak) [reply test/widgets/cards/history_info_card_test.dart]: done

**Review by quiteachaos — APPROVED** [review]

**Pull request merged**